src/storage/mongodb/ChargingStationStorage.ts

```diff
@@ -707,6 +707,33 @@ export default class ChargingStationStorage {
       });
     }
 
+    if (params.withChargingStation || params.withSiteArea) {
+      // Charging Stations
+      DatabaseUtils.pushChargingStationLookupInAggregation({
+        tenantID, aggregation, localField: 'chargingStationID', foreignField: '_id',
+        asField: 'chargingStation', oneToOneCardinality: true, oneToOneCardinalityNotNull: false
+      });
+      // Site Areas
+      DatabaseUtils.pushSiteAreaLookupInAggregation({
+        tenantID, aggregation, localField: 'chargingStation.siteAreaID', foreignField: '_id',
+        asField: 'chargingStation.siteArea', oneToOneCardinality: true, oneToOneCardinalityNotNull: false
+      });
+      // Convert
+      DatabaseUtils.pushConvertObjectIDToString(aggregation, 'chargingStation.siteAreaID');
+      DatabaseUtils.pushConvertObjectIDToString(aggregation, 'chargingStation.siteArea.siteID');
+      // Check Site ID
+      if (!Utils.isEmptyArray(params.siteIDs)) {
+        // Build filter
+        aggregation.push({
+          $match: {
+            'chargingStation.siteArea.siteID': {
+              $in: params.siteIDs.map((id) => id)
```

Comment:
How did you manage to test when the Side ID is an ObjectID and the conversion here is missing?
I did the change but you have to test it again:
![image](https://user-images.githubusercontent.com/25266945/89439352-5ca39e80-d74a-11ea-97d7-0efe7c4be801.png)
Let me know